### PR TITLE
Sub-Task No 2: Test & Fix Sales order (w/ ship ) - ProductPlanning - …

### PR DESCRIPTION
**Pull request by dm-ps**
…Manufacturing

refs: https://github.com/metasfresh/metasfresh/issues/13717

### CODE REVIEW

File: backend/de.metas.manufacturing/src/main/java/org/eevolution/model/MPPOrder.java [hunk 1]
```diff
@@ -442,10 +442,11 @@ public boolean reActivateIt()
 		final PPOrderId orderId = PPOrderId.ofRepoId(getPP_Order_ID());
 		orderBOMsRepo.markBOMLinesAsNotProcessed(orderId);
 
-		ModelValidationEngine.get().fireDocValidate(this, ModelValidator.TIMING_AFTER_REACTIVATE);
-
 		setDocAction(IDocument.ACTION_Complete);
 		setProcessed(false);
+
+		ModelValidationEngine.get().fireDocValidate(this, ModelValidator.TIMING_AFTER_REACTIVATE);
```
Comment (metas-ts) [inline backend/de.metas.manufacturing/src/main/java/org/eevolution/model/MPPOrder.java:448]: 👍 checked with MOrder and MInvoice..

File: backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Order_PostMaterialEvent.java [hunk 2]
```diff
@@ -79,16 +81,21 @@ public void fireMaterialEvent_deletedPPOrder(
...
+	public void postMaterialEvent_ppOrderDocStatusChange(@NonNull final I_PP_Order ppOrderRecord, @NonNull final DocTimingType docTimingType)
 	{
-		final PPOrderChangedEvent changeEvent = PPOrderChangedEventFactory
+		PPOrderChangedEvent changeEvent = PPOrderChangedEventFactory
 				.newWithPPOrderBeforeChange(ppOrderConverter, ppOrderRecord)
 				.inspectPPOrderAfterChange();
 
+		if (docTimingType == DocTimingType.AFTER_REACTIVATE)
+		{
+			changeEvent = changeEvent.withNewDocStatus(DocStatus.InProgress);
```
Comment (metas-ts) [inline backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Order_PostMaterialEvent.java:96]: if the new doctype can't always be added to the event, please add a comment on why that is

**Review by metas-ts — CHANGES_REQUESTED** [review]
- please check my minor remark
- please add test-IDs
- in https://github.com/metasfresh/metasfresh/issues/13663 daniela observerd that we have no tests for a shipment getting reactivated or reversed
  please make sure that we have such a coverage in this subissue

File: backend/de.metas.manufacturing/src/main/java/org/eevolution/model/MPPOrder.java (hunk 1 above)
Comment (dm-ps) [reply backend/de.metas.manufacturing/src/main/java/org/eevolution/model/MPPOrder.java:448]: Regarding Daniela's observation for #13663, I extended the first scenario in `src/test/resources/de/metas/cucumber/features/materialCockpit/materialCockpit_SO_noProductPlanning.feature`, test case with `@Id:S0189_100`

**Pull request merged**